Lesson2

JustSenoko opened the pull request:
Задание к уроку №2

JustSenoko committed:
добавлен lesson2
класс Application и 2 класса с исключениями

JustSenoko committed:
Create README.txt

usharik commented on src/ru/geekbrains/lesson2/Application.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    private static int sumArray(String[][] arr) throws MyArraySizeException, MyArrayDataException {
+        int requiredSize = 4;
```
Такие константы лучше объявлять в виде static final полей класса.

usharik commented on src/ru/geekbrains/lesson2/Application.java:
```diff
@@ -0,0 +1,58 @@
+                try {
+                    sum += Integer.parseInt(arr[i][j]);
+                } catch (NumberFormatException ex) {
+                    throw new MyArrayDataException(i, j);
```
Можно было бы еще добавить параметр в конструктор, для передачи исключения, которое привело к MyArrayDataException

JustSenoko committed:
требуемый размер массива вынесен в поле класса
в вызов конструктора исключения MyArraySizeException добавлен тип исходного исключения
